processor/src/main/java/org/mapstruct/ap/internal/model/common/SourceRHS.java

```diff
@@ -149,6 +149,9 @@ public Type getSourceTypeForMatching() {
             else if ( sourceType.isStreamType() ) {
                 return first( sourceType.determineTypeArguments( Stream.class ) );
             }
+            else if ( sourceType.isArrayType() ) {
+                return sourceType.getComponentType();
+            }
```

Comment:
I think that we might need to have another `else if ( sourceType.isIterableType() )` here as well.

Reply:
you are correct :+1: